libs/cdk/src/lib/template-management/template-manager.ts

```diff
@@ -0,0 +1,168 @@
+import { EmbeddedViewRef, TemplateRef } from '@angular/core';
+import { combineLatest, EMPTY, isObservable, Observable, of } from 'rxjs';
+import {
+  catchError,
+  distinctUntilChanged,
+  filter,
+  map,
+  merge as mergeWith,
+  switchAll,
+  switchMap,
+  withLatestFrom,
+} from 'rxjs/operators';
+import { onStrategy, rxMaterialize } from '../render-strategies/utils';
+import { RenderWork, RxNotification, RxNotificationKind } from '../render-strategies/model';
+import {
+  RenderSettings,
+  RxBaseTemplateNames,
+  RxViewContext,
+  TemplateSettings,
+} from './model';
+import {
+  getEmbeddedViewCreator,
+  getTNode,
+  notificationKindToViewContext,
+  notifyAllParentsIfNeeded,
+  notifyInjectingParentIfNeeded,
+  strategyHandling,
+  templateHandling,
+  templateTriggerHandling,
+  TNode,
+} from './utils';
+import { CoalescingOptions } from '../model';
+
+export interface RenderAware<T> {
+  nextStrategy: (nextConfig: string | Observable<string>) => void;
+  render: (values$: Observable<T>) => Observable<any>;
+}
+
+export interface TemplateManager<
+  T,
+  C extends RxViewContext<T>,
+  N = RxBaseTemplateNames | string
+> extends RenderAware<T> {
+  addTemplateRef: (name: N, templateRef: TemplateRef<C>) => void;
+  addTrigger: (trigger$: Observable<RxNotification<T>>) => void;
+}
+
+export function createTemplateManager<
+  T,
+  C extends RxViewContext<T>,
+  N = RxBaseTemplateNames | string
+>(config: {
+  renderSettings: RenderSettings<T, C>;
+  templateSettings: TemplateSettings<T, C>;
+  templateTrigger$?: Observable<RxNotificationKind>;
+  notificationToTemplateName: {
+    [rxKind: string]: (
+      value: T,
+      templates: { get: (name: N) => TemplateRef<C> }
+    ) => N;
+  };
```

Comment:
can we make a `type` out of it?
 suggestion: 
```ts

Record<RxNotificationKind, (
      value: T,
      templates: { get: (name: N) => TemplateRef<C> }
    ) => N>
```

Reply:
Done